### PR TITLE
add wkb encoder and decoder

### PR DESCRIPTION
**Pull request by FluorineDog**

### CODE REVIEW

File: cpp/src/gis/cuda/common/gis_definitions.cpp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  assert(data_state_ == DataState::PrefixSumOffset_EmptyData);
+  assert(tags_.size() == gpu_ctx.size);
+  GpuMemcpy(metas_.data(), gpu_ctx.metas, metas_.size());
+  GpuMemcpy(values_.data(), gpu_ctx.values, values_.size());
+  data_state_ = DataState::PrefixSumOffset_FullData;
+}
+
+// only for testing
+// create Geometry from WktArray
+namespace GeometryVectorFactory {
```
Comment (GuoRentong) [inline cpp/src/gis/cuda/common/gis_definitions.cpp]: move GeometryVectorFactory to test dir

File: cpp/src/gis/cuda/common/gis_definitions.cpp [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  tags_.clear();
+  metas_.clear();
+  values_.clear();
+  meta_offsets_.clear();
+  value_offsets_.clear();
+  data_state_ = DataState::Invalid;
+}
+
+// only for testing
+bool test_cuda_abi(const std::string& str) {
```
Comment (GuoRentong) [inline cpp/src/gis/cuda/common/gis_definitions.cpp]: move test_cuda_abi to test dir

File: cpp/src/gis/cuda/conversion/arrow_decode.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace zilliz {
+namespace gis {
+namespace cuda {
+
+std::shared_ptr<arrow::Array> GeometryVectorToArrowWkb(const GeometryVector&);
+
+using internal::WkbArrowContext;
+
+GeometryVector ArrowWkbToGeometryVector(const std::shared_ptr<arrow::Array>& wkb_) {
```
Comment (GuoRentong) [inline cpp/src/gis/cuda/conversion/arrow_decode.cpp]: do not end params with _
consider other names such as arrow_wkb

File: cpp/src/gis/cuda/common/gis_definitions.cpp (hunk 2 above)
Comment (FluorineDog) [reply cpp/src/gis/cuda/common/gis_definitions.cpp]: delete it since no use from the beginning.

File: cpp/src/gis/cuda/common/gis_definitions.h [hunk 4]
```diff
@@ -24,16 +24,24 @@
 #include <string>
 #include <tuple>
 #include <vector>
-using std::vector;
+
+#include "gis/cuda/mock/arrow/api.h"
 template <typename T>
-using GpuVector = vector<T>;  // TODO: use gpu vector, now just placeholder
+using GpuVector = std::vector<T>;  // TODO: use gpu vector, now just placeholder
```
Comment (GuoRentong) [inline cpp/src/gis/cuda/common/gis_definitions.h]: replace std::vector with real gpu implementation with another pr

**Pull request merged**